src/lib/memoise.ts

```diff
@@ -0,0 +1,16 @@
+export const memoiseAsync = <T>(fn: () => Promise<T>): [() => void, () => Promise<T>] => {
+    let res: T | undefined;
+
+    const retFn = async () => {
+        if (res !== undefined) {
+            return res;
+        }
+
+        res = await fn();
+        return res;
+    };
+
+    const resetFn = () => (res = undefined);
```

Comment:
Return type here too?

Reply:
Generally I wouldn't add typing to a nested function unless the type was really unclear. Here I'm not sure it would add much as the function itself is trivial and also the wrapping function has a return type that captures this info.